subprojects/language-java/src/integTest/groovy/org/gradle/api/tasks/compile/JavaCompileIntegrationTest.groovy

```diff
@@ -905,6 +907,29 @@ class JavaCompileIntegrationTest extends AbstractIntegrationSpec {
         succeeds "-Pjava8", "clean", "compileJava"
     }
 
+    // bootclasspath has been removed in Java 9+
+    @Requires(TestPrecondition.JDK8_OR_EARLIER)
+    @Issue("https://github.com/gradle/gradle/issues/19817")
+    def "nags if bootclasspath is provided as a path instead of a single file"() {
+        def jre = AvailableJavaHomes.getBestJre()
+        def bootClasspath = TextUtil.escapeString(jre.absolutePath) + "/lib/rt.jar${File.pathSeparator}someotherpath"
+        buildFile << """
+            plugins {
+                id 'java'
+            }
+            tasks.withType(JavaCompile) {
+                options.bootstrapClasspath = project.layout.files("$bootClasspath")
+            }
+        """
+        file('src/main/java/Foo.java') << 'public class Foo {}'
+
+        expect:
+        executer.expectDeprecationWarning("FileCollection 'file collection' that is being mapped to a path contains already concatenated files instead of a collection of single files." +
+            " This can lead to uncontrolled failures. Problematic concatenations are: ${Paths.get(bootClasspath)}." +
+            " This will fail with an error in Gradle 8.0. Add files to a collection as single files instead of manually concatenating them.")
```

Comment:
```suggestion
        executer.expectDeprecationWarning("Converting files to a classpath string when their paths contain the path separator %s has been deprecated." +
            " The path separator is not a valid element of a file path. Problematic paths in 'file collection' are '${Paths.get(bootClasspath)}'." +
            " This will fail with an error in Gradle 8.0. Add the individual files to the file collection instead.")
```

How about something like ☝️ ?

Reply:
Works for me